### PR TITLE
Don't look for fmemopen if the macOS version is less than 10.13

### PR DESCRIPTION
**Pull request by datalogics-kam**
- Leptonica uses CheckFunctionExists to see if it's there, but that only checks
  if a call to the function can be linked. That's dependent on the SDK, but not
  the deployment target.

- It should use CheckSymbolExists, and use the header, and elevate the
  availability warning to an error, but that's a more complicated fix.

- So just don't look.

- Fixes problems linking products when building under certain macOS
  environments, when fmemopen should not be used due to the deployment target.

### CODE REVIEW

File: mkenv.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    install_project_requirements(output_route)
+
+
+def windows():
+    'returns True on Windows platforms'
+    return platform.system() == 'Windows'
+
+
+def mac():
+    'returns True on Mac OS X'
```
Comment (datalogics-robb) [inline mkenv.py:113]: mac() and aix() aren't used, and when they were used, it was to work around a specific bug.  I would like to see them removed.

**Review by datalogics-robb — CHANGES_REQUESTED** [review]
The versions of jpeg-dl and tiff-dl  seem to be out of date with the rest of APDFL.

File: mkenv.py (hunk 1 above)
Comment (datalogics-kam) [reply mkenv.py:113]: I copied them from `ocrbox`. Unfortunately, I haven't got around to finishing the cleanup of mkenv there.

**Comment by datalogics-kam** [comment]
>The versions of jpeg-dl and tiff-dl seem to be out of date with the rest of APDFL.

I'll update them, but they're not even used any more (`ocrbox` turns off everything but PNG here: https://octocat.dlogics.com/datalogics/ocrbox/blob/develop/conanfile.py#L20).

**Review by datalogics-robb — APPROVED** [review]

**Pull request merged**